src/marqo/core/document/tensor_fields_container.py

```diff
@@ -0,0 +1,342 @@
+
+        for field_name, tensor_content in doc.items():
+            if not tensor_content.is_tensor_field:
+                # If this is not top level tensor field, we do not populate from existing tensor
+                # TODO confirm if this is expected for unstructured as well
+                continue
+
+            if tensor_content.embeddings:
+                # Already populated, might be a custom vector
+                continue
+
+            if field_name in existing_multimodal_weights:
+                # for multimodal_combo fields
+
+                if tensor_content.field_type != FieldType.MultimodalCombination:
+                    # Field with the same name is not a multimodal field in this batch
+                    continue
+
+                weights = cast(MultiModalTensorFieldContent, tensor_content).weights
+                if existing_multimodal_weights[field_name] != weights:
+                    # mapping config is different, need to re-vectorise
+                    continue
+
+                if any([sub_field not in existing_marqo_doc or sub_field not in doc or
+                        existing_marqo_doc[sub_field] != doc[sub_field].field_content for sub_field in weights.keys()]):
+                    # If content of any subfields does not match
+                    continue
+
+            else:
+                # for other tensor fields
+
+                if field_name not in existing_marqo_doc:
+                    # This is a new field added to the doc, we need to vectorise it
+                    continue
+
+                if existing_marqo_doc[field_name] != tensor_content.field_content:
+                    # Field content has changed, we need to re-vectorise
+                    continue
+
+            if (constants.MARQO_DOC_TENSORS not in existing_marqo_doc or
+                    field_name not in existing_marqo_doc[constants.MARQO_DOC_TENSORS]):
+                # This field is not a tensor field in existing doc, we need to vectorise
+                continue
+
+            existing_tensor = existing_marqo_doc[constants.MARQO_DOC_TENSORS][field_name]
+            tensor_content.populate_chunks_and_embeddings(existing_tensor[constants.MARQO_DOC_CHUNKS],
+                                                          existing_tensor[constants.MARQO_DOC_EMBEDDINGS])
+
+    def collect(self, doc_id: str, field_name: str, field_content: Any, text_field_type: Optional[FieldType]) -> Any:
+        if field_name not in self._tensor_fields and field_name not in self._multimodal_sub_field_reverse_map:
+            # not tensor fields, no need to collect
+            return field_content
+
+        if self.is_custom_tensor_field(field_name):
+            content = field_content['content']
+            embedding = field_content['vector']
+            tensor_field_content = TensorFieldContent(
+                field_content=content,
+                field_type=FieldType.CustomVector,
+                is_tensor_field=True,
```

Comment:
When is this flag false?

Reply:
subfields of the Multimodal fields that are not top level tensor fields